src/CSnakes.Runtime/Locators/ManagedPythonLocator.cs

```diff
@@ -0,0 +1,215 @@
+
+            // Download and extract the Zstd tarball
+            logger.LogInformation("Downloading Python from {DownloadUrl}", downloadUrl);
+            string tempFilePath = DownloadFileToTempDirectoryAsync(downloadUrl).GetAwaiter().GetResult();
+            string tarFilePath = DecompressZstFile(tempFilePath);
+            ExtractTar(tarFilePath, downloadPath, logger);
+            logger.LogInformation("Extracted Python to {downloadPath}", downloadPath);
+
+            // Delete the tarball and temp file
+            File.Delete(tarFilePath);
+            File.Delete(tempFilePath);
+        }
+        catch (Exception ex)
+        {
+            logger.LogError(ex, "Failed to download and extract Python");
+            // If the install failed somewhere, delete the folder incase it's half downloaded
+            if (Directory.Exists(installPath))
+            {
+                Directory.Delete(installPath, true);
+            }
+
+            throw;
+        }
+        finally
+        {
+            // Delete the lock file
+            File.Delete(lockfile);
+        }
+        return LocatePythonInternal(installPath);
+    }
+
+    protected override string GetLibPythonPath(string folder, bool freeThreaded = false)
+    {
+        string suffix = freeThreaded ? "t" : "";
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+        {
+            return Path.Combine(folder, $"python{Version.Major}{Version.Minor}{suffix}.dll");
+        }
+
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+        {
+            return Path.Combine(folder, "lib", $"libpython{Version.Major}.{Version.Minor}{suffix}.dylib");
+        }
+
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+        {
+            return Path.Combine(folder, "lib", $"libpython{Version.Major}.so");
+        }
+
+        throw new PlatformNotSupportedException($"Unsupported platform: '{RuntimeInformation.OSDescription}'.");
+    }
+
+    private static async Task<string> DownloadFileToTempDirectoryAsync(string fileUrl)
+    {
+        using HttpClient client = new();
+        using HttpResponseMessage response = await client.GetAsync(fileUrl);
+        response.EnsureSuccessStatusCode();
+
+        string tempFilePath = Path.GetTempFileName();
+        using FileStream fileStream = new FileStream(tempFilePath, FileMode.Create, FileAccess.Write, FileShare.None);
+        await response.Content.CopyToAsync(fileStream);
+
+        return tempFilePath;
+    }
+
+    private static string DecompressZstFile(string zstFilePath)
+    {
+        string tarFilePath = Path.ChangeExtension(zstFilePath, ".tar");
+        using (var inputStream = new FileStream(zstFilePath, FileMode.Open, FileAccess.Read))
+        using (var decompressor = new DecompressionStream(inputStream))
+        using (var outputStream = new FileStream(tarFilePath, FileMode.Create, FileAccess.Write))
+        {
+            decompressor.CopyTo(outputStream);
+        }
```

Comment:
```suggestion
        using var inputStream = new FileStream(zstFilePath, FileMode.Open, FileAccess.Read);
        using var decompressor = new DecompressionStream(inputStream);
        using var outputStream = new FileStream(tarFilePath, FileMode.Create, FileAccess.Write);
        decompressor.CopyTo(outputStream);
```

Just so it's consistent with how other `using`'s are done.